source/slang/slang-lower-to-ir.cpp

```diff
@@ -1919,6 +1919,28 @@ struct ValLoweringVisitor : ValVisitor<ValLoweringVisitor, LoweredValInfo, Lower
         return LoweredValInfo::simple(getBuilder()->getIntValue(type, val->getValue()));
     }
 
+    IRType* visitDifferentialPairType(DifferentialPairType* pairType)
+    {
+        IRType* primalType = lowerType(context, pairType->getPrimalType());
+        if (as<IRAssociatedType>(primalType) || as<IRThisType>(primalType))
+        {
+            List<IRInst*> operands;
+            SubstitutionSet(pairType->getDeclRef())
+                .forEachSubstitutionArg(
+                    [&](Val* arg)
+                    {
+                        auto argVal = lowerVal(context, arg).val;
+                        SLANG_ASSERT(argVal);
+                        operands.add(argVal);
+                    });
+
+            auto undefined = getBuilder()->emitUndefined(operands[1]->getFullType());
```

Comment:
This is another case where things would be much cleaner if DifferentialPair is just an associated type, and we can simply lower this normally as other associated types. The `undefined` operand for witness just feels wrong, and is suggesting our current handling of DifferentialPair isn't correct.

Reply:
Yup, strong agree.. I'm adding this point to the "autodiff 2.0 type system" overhaul.